.github/workflows/continuous-deployment-server.yml

```
@@ -0,0 +1,98 @@
+name: Continuous Deployment indy-vdr-proxy-server
+
+on:
+  push:
+    branches: [main]
+
+  workflow_dispatch:
+
+env:
+  DH_USERNAME: ${{ secrets.DOCKER_HUB_LOGIN }}
+  DH_TOKEN: ${{ secrets.DOCKER_HUB_PWD }}
+  IMAGE_NAME: "indy-vdr-proxy-server"
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Setup node v22
+        uses: actions/setup-node@v4
+        with:
+          node-version: 22
+
+      - name: Install dependencies
+        run: |
+          cd packages/server
+          yarn install
+
+      - uses: codfish/semantic-release-action@v3
+        id: semantic
+        with:
+          branches: |
+            [
+              'main',
+              { 
+                name: 'dev',
+                prerelease: true
+              }
+            ]
+          plugins: |
+            [
+              [
+                "@semantic-release/commit-analyzer",
+                {
+                  "preset": "angular",
+                  "releaseRules": [
+                    {
+                      "type": "refactor",
+                      "release": "patch"
+                    }
+                  ]
+                }
+              ],
+              [
+                "@semantic-release/npm",
+                {
+                  "npmPublish": false
+                }
+              ],
+              "@semantic-release/release-notes-generator",
+            ]
+
+      - name: Log in to Docker Hub
+        if: steps.semantic.outputs.new-release-published == 'true'
+        run: |
+          echo "$DH_TOKEN" | docker login -u "$DH_USERNAME" --password-stdin
+
+      - name: Build Docker image
+        if: steps.semantic.outputs.new-release-published == 'true'
+        run: |
+          cd packages/server
+          docker build -f Dockerfile -t $DH_USERNAME/$IMAGE_NAME:${GITHUB_REF##*/} .
```

```suggestion
          docker build -f Dockerfile -t $DH_USERNAME/$IMAGE_NAME:dev .
```

We can simply use `dev` tag for the moment, until we integrate all these steps with the regular server CI mechanism (the one that is currently using release-please).